analyticsservice/analytics.go

```diff
@@ -43,13 +45,17 @@ type analyticsEvent struct {
 
 // AnalyticsService provides a way to cache and send analytics to the server
 type AnalyticsService struct {
-	mx            *sync.Mutex
-	analyticsChan chan analyticsEvent
-	analyticsData map[string]analyticsEvent
-	timeout       time.Duration
-	logger        logger.Logger
-	metricsClient *metricsclient.ClientWithResponsesInterface
-	environmentID string
+	analyticsChan          chan analyticsEvent
+	evaluationsAnalyticsMx *sync.Mutex
+	targetAnalyticsMx      *sync.Mutex
+	seenTargetsMx          *sync.RWMutex
```

Comment:
Opted for a RWMutex here because the seen targets cache should mostly be reads once it's filled up.  It's possible a user could send all unique targets, but it is more likely in my estimation that targets will be reused.